### DISCUSSION TITLE
How to use a custom embedding model with llama_index on supabase

### DISCUSSION THREAD

**Discussion by projects-g** [supabase/supabase#18868] [Questions]
Since different models use diff dimensions for embeddings, not using the OpenAI embeddings throws the following error :

```expected 1536 dimensions, not 768```

Following one of your template examples : 
Source : https://colab.research.google.com/github/supabase/supabase/blob/master/examples/ai/llamaindex/llamaindex.ipynb#scrollTo=ce9bc85b-e844-407c-a0ad-ccf6af3c8866

**Comment by jnorris441** [comment]
@projects-g  you will need to alter your vector column with the correct number of dimensions:

```
alter table documents
alter column embedding
type vector (768)
```

**Comment by projects-g** [comment]
@jnorris441 Thanks !

It is designed in a way that automatically creates the DB (uses OpenAI embedding dimension and then we have to update the DB). An alternative is also to define the embedding model when we are creating the Supabase index using the ServiceContext.

        self.service_context = ServiceContext.from_defaults(
            embed_model="local:BAAI/bge-base-en-v1.5",
        )
